torch/testing/_internal/common_utils.py

```diff
@@ -2376,27 +2376,36 @@ def noncontiguous_like(t):
         return t
 
     # Special-cases 0-dim tensors
-    if t.ndim == 0:
-        result = t.detach().unsqueeze(0).repeat_interleave(2, dim=-1)
-        if t.dtype.is_floating_point or t.dtype.is_complex:
-            result[0] = math.nan
-        else:
-            result[0] = 0
-        result.set_(result.storage(), 1, t.size(), ())
-        result.requires_grad_(t.requires_grad)
-        return result
-
-    # 1+ dim tensor case
+    zero_dim = t.ndim == 0
+    if zero_dim:
+        t = t.unsqueeze(0)
+
+    # Handle conj bit. repeat_interleave resolves the conj, which is a bit annoying.
```

Comment:
Nice catch

Reply:
Actually I don't see any reason why we need to resolve conjugation or negation for `repeat_interleave`. We should register the `repeat_interleave` as fallthrough for both `Conjugate` and `Negative` dispatch key and simply unset the conj/neg bit for `self` before the computation and then reset it.  `repeats` is guaranteed to be an int tensor so we don't have to worry about it.

Reply:
It might be worth making this change in this PR so that you don't have to add special conj/neg handling